src/thir.md

```diff
@@ -19,8 +20,10 @@ the types have been filled in, which is possible after type checking has complet
 But it has some other interesting features that distinguish it from the HIR:
 
 - Like the MIR, the THIR only represents bodies, i.e. "executable code"; this includes
-  function bodies, but also `const` initializers, for example. Consequently, the THIR
-  has no representation for items like `struct`s or `trait`s.
+  function bodies, but also `const` initializers, for example. Specifically, everything with a
+  `DefId` can (and does) have a THIR body created. Consequently, the THIR
+  has no representation for items like `struct`s or `trait`s. References to those items are
+  represented as `HirId`s in the THIR.
```

Comment:
This isn't actually true, `struct`s and `trait`s do have `DefId`s, and the THIR (like the HIR) references them through a `DefId`. If you want to link to a list of all things that have a body you could use [`BodyOwnerKind`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/enum.BodyOwnerKind.html).